source/getting-started/control-with-amdc/encoder-fb/index.md

```diff
@@ -1,13 +1,97 @@
+
+### Obtaining encoder count and translating it into rotor position
+
+<img src="./resources/EncoderCodeBlockDiargam.svg" width="100%" align="center"/>
+
+As a first step, the user may use the AMDC `drv/encoder` driver module `encoder_get_position()` to get the count of the encoder reading. The`drv/encoder` driver module also has a function called `encoder_get_steps()` which gives the incremental change in the encoder position. Whereas, `encoder_get_position()` gives the actual position of the shaft and this can be converted the rotor position in radians. Next, the user needs to verify if the encoder count is increasing or decreasing with counter-clock wise rotation of shaft. This may be done by manually rotating the shaft and observing the trend of the reported position with respect to the direction of rotation. This document follows the convention of a positive rotor angle in the counter clockwise (CCW) direction of shaft rotation. Using this information, along with the offset and total encoder counts per revolution, the obtained count can be translated into angular position using a simple linear equation. The user must ensure that angle is within the bounds of 0 and 2$\pi$ by appropriately wrapping the variable using the `mod` function.
+
+Example code to convert encoder to angular position in radians:
+```C
+double task_get_theta_m(void)
+{
+    // Get raw encoder position
+    uint32_t position;
+    encoder_get_position(&position);
+
+    int ENCODER_COUNT_PER_REV, CCW;
+    double enc_theta_m_offset;
+
+    // User to set encoder count per revolution
+    ENCODER_COUNT_PER_REV = 1024;
+
+    // Set 1 if encoder count increases with CCW rotation of shaft, Set 0 if encoder count increases with CW rotation of shaft
+    CCW = 1; 
+
+    // Angular position to be computed
+    double theta_m_enc;
+
+    // User to set encoder offset
+    enc_theta_m_offset = 100;
+
+
+    // Convert to radians
+    if (CCW){
+        theta_m_enc = (double) PI2 * ( ( (double)position - enc_theta_m_offset )/ (double) ENCODER_COUNT_PER_REV);
+    }
+    else{
+        theta_m_enc = (double) PI2 * ( ( (double)ENCODER_COUNT_PER_REV - (double)1 -(double)position + enc_theta_m_offset )/ (double) ENCODER_COUNT_PER_REV);
+    }
+    
+     // Mod by 2 pi
+    theta_m_enc = fmod(theta_m_enc,PI2 );
+    return theta_m_enc;
+}
+```
+
+
+
+### Finding the offset
+
+The example code shown above makes uses of an encoder offset value. It is necessary for the user to find this offset experimentally for their machine. For synchronous machines, this offset is the count value measured by the encoder when the d-axis of the rotor is aligned with the phase U winding axis of the stator. 
```

Comment:
Based on how you introduce offset above (see my comment above) you might need to slightly tweak this text